util/ranger/detacher.go

```diff
@@ -112,7 +112,7 @@ func getEqOrInColOffset(expr expression.Expression, cols []*expression.Column) i
 			offset = curOffset
 		}
 		return offset
-	case ast.EQ:
+	case ast.EQ, ast.NullEQ:
```

Comment:
We have to specially check if it is the `a <=> null` case here, because as I mentioned in the PR description of https://github.com/pingcap/tidb/pull/18315:

> I tried to handle IsNull as well, i.e, match a = 1 and b is null and c > 1 to composite index key(a, b, c), but found out that IsNull is treated as range scan instead of point get on purpose to return correct result when nullable unique index has multiple matches for filter x is null(see also #7163). We have to prepare a lot of changes before we can treat IsNull as equal access conditions for composite indexes, so I give it up for now.